tests/system/libraries/SystemTestSpy/speechSpyGlobalPlugin.py

```diff
@@ -68,7 +68,9 @@ def _registerWithExtensionPoints(self):
 
 	# callbacks for extension points
 	def _onNvdaStartupComplete(self):
-		self._isNvdaStartupComplete = True
+		# Queue the setting of the completion variable,
+		# To ensure that NvDA's core loop has started running, and it has processed initial focus.
+		queueHandler.queueFunction(queueHandler.eventQueue, lambda: setattr(self, '_isNvdaStartupComplete', True))

```

Comment:
Perhaps the code at `source/core.py:564: postNvdaStartup.notify()` should be queued instead? If we are saying the loop must have started before NVDA's startup is complete, then the `postNvdaStartup` action is incorrect.

Reply:
I think for system tests it makes sense to ensure the loop has started and initial focus is processed. But for other uses I'm not sure. I'm more inclined just to change the system tests for now, rather than change the behaviour of that action, which other 3rd party code *could* be using.